data-catalog-api/server/service/src/main/java/org/apache/airavata/datacatalog/api/service/DataCatalogAPIService.java

```diff
@@ -194,20 +194,53 @@ public void searchDataProducts(DataProductSearchRequest request,
             StreamObserver<DataProductSearchResponse> responseObserver) {
 
         try {
+            // takeout all the DataProduct
             MetadataSchemaQueryResult searchResult = dataCatalogService.searchDataProducts(request.getUserInfo(),
                     request.getSql());
-            List<DataProduct> dataProducts = searchResult.dataProducts();
-            responseObserver.onNext(DataProductSearchResponse.newBuilder().addAllDataProducts(dataProducts).build());
+            List<DataProduct> allDataProducts = searchResult.dataProducts();
```

Comment:
Better not to load all the DPs at once. If there are 1000s of dps match with the query then this would cause memory issues.
This should be handle at query level